core/src/module/marine_module.rs

```diff
@@ -251,33 +251,40 @@ impl<WB: WasmBackend> MModule<WB> {
     fn add_host_imports(
         store: &mut <WB as WasmBackend>::Store,
         linker: &mut <WB as WasmBackend>::Imports,
-        raw_imports: HashMap<String, RawImportCreator<WB>>,
-        host_imports: HashMap<String, HostImportDescriptor<WB>>,
+        raw_imports: HashMap<u32, HashMap<String, RawImportCreator<WB>>>,
+        host_imports: HashMap<u32, HashMap<String, HostImportDescriptor<WB>>>,
         mit: &MITInterfaces<'_>,
     ) -> MResult<()> {
         use crate::host_imports::create_host_import_func;
+        for (version, raw_imports) in raw_imports {
+            let namespace = version_to_host_namespace(version);
+            let raw_imports = raw_imports
+                .into_iter()
+                .map(|(name, creator)| (name, creator(store.as_context_mut())))
+                .collect::<Vec<_>>();
+
+            linker.register(store, &namespace, raw_imports)?;
```

Comment:
```suggestion
            linker.register(store, namespace, raw_imports)?;
```

to avoid recreating the string again